ArduPlane/AP_Arming.cpp

```diff
@@ -321,6 +321,15 @@ bool AP_Arming_Plane::arm(const AP_Arming::Method method, const bool do_arming_c
     // rising edge of delay_arming oneshot
     delay_arming = true;
 
+#if MODE_AUTOLAND_ENABLED
+    //capture heading for autoland mode if option is set and using a compass
+    if (plane.ahrs.use_compass() && plane.tkoff_option_is_set(AP_FixedWing::TakeoffOption::AUTOLAND_DIR_ON_ARM)) {
+        plane.takeoff_state.initial_direction.heading = wrap_360(plane.ahrs.yaw_sensor * 0.01f);
```

Comment:
I would have expected `landing_dir_off` to be added here as with the ground course method.

Reply:
offset should not be added to the armed set....the user knows the exact direction he wants